kbase-extension/static/kbase/js/widgets/function_output/codeCellLauncher.js

```diff
@@ -0,0 +1,121 @@
+        options: {
+            output: null,
+            workspaceID: null,
+            loadingImage: Config.get('loading_gif')
+        },
+
+        init: function(options) {
+            this._super(options);
+            // Create a message pane
+            this.$messagePane = $('<div/>').addClass('kbwidget-message-pane kbwidget-hide-message');
+            this.$elem.append(this.$messagePane);
+            return this;
+        },
+
+        launchCodeCell: function (code) {
+            const thisCell = this.$elem[0].parentElement.parentElement.kb_obj.cell,
+                cellId = utils.getMeta(thisCell, 'attributes', 'id'),
+                cellIndex = Jupyter.notebook.find_cell_index(thisCell),
+                newCellId = new Uuid(4).format(),
+                setupData = {
+                    type: 'code',
+                    cellId: newCellId,
+                    parentCellId: cellId,
+                };
+
+            let newCell = Jupyter.notebook.insert_cell_below('code', cellIndex, setupData);
+            newCell.set_text(code);
+            newCell.execute();
+
+            // Don't want duplicate child cells on load. This is the best way I can think to do it right now
+            Jupyter.notebook.delete_cell(cellIndex)
+        },
+
+        get_python_code: function (type, matrix_ref) {
+            let code;
+            if (type === 'dataframe') {
+                code = 'from biokbase.narrative.viewers import get_df\n' +
+                    `df = get_df('${matrix_ref}')\ndf`
+            } else if (type === 'clustergrammer') {
+                code = 'from biokbase.narrative.viewers import view_as_clustergrammer\n' +
+                    `view_as_clustergrammer('${matrix_ref}')`
+            } else if (type === 'seaborn box') {
+                code = 'import seaborn as sns\n' +
+                    'from biokbase.narrative.viewers import get_df\n' +
+                    `df = get_df('${matrix_ref}', None, None)\n` +
+                    'sns.boxplot(data=df).set(xlabel="Columns", ylabel="Row Values");'
+            } else if (type === 'plotly stackedbars') {
+                code = 'import plotly\n' +
+                    'import plotly.graph_objs as go\n' +
+                    'from biokbase.narrative.viewers import get_df\n\n' +
+                    `df = get_df('${matrix_ref}', None, None)\n` +
+                    'plotly.offline.init_notebook_mode(connected=True)\n' +
+                    'plotly.offline.iplot({\n' +
+                    '    "data": [go.Bar(x=df.columns, y=row, name=i) for i, row in df.iterrows()],\n' +
+                    '    "layout": go.Layout(title="Stacked bars" xaxis={"title": "Columns"}, yaxis={"title": "Row Values"}, barmode="stack")\n' +
+                    '})'
+
+            } else {
+                this.showMessage(`Unsupported output type ${type}`);
+            }
+            return code;
+        },
+
+        loggedInCallback: function(event, auth) {
```

Comment:
You aren't really doing anything that needs authentication here. Might as well just put this logic in the init function.